### PR TITLE
prepare era and sample specific configs

### PR DESCRIPTION
**Pull request by swozniewski**

### CODE REVIEW

File: config/config.py [hunk 1]
```diff
@@ -112,6 +118,10 @@ def build_config():
         ],
     }
 
+    config["producer_mods"] = [
```
Comment (harrypuuter) [inline config/config.py:121]: I would stick with writing out all names, so 

RemoveProducer instead of Prod_Remove
AppendProducer instead of Prod_Append

Comment (harrypuuter) [reply config/config.py:121]: also `producer_modifications` instead of `producer_mods`

File: config/utility.py [hunk 2]
```diff
@@ -13,3 +16,80 @@ def AddSystematicShift(
...
+
+class ProducerRule:
+    def __init__(self, producers, samples, scopes, invert=False, update_output=True):
+        self.producers = producers
+        self.samples = samples
+        self.scopes = scopes
+        self.invert = invert
+        self.update_output = update_output
+
+    def applicable(self, sample):
```
Comment (harrypuuter) [inline config/utility.py:68]: what does this function do ?

File: config/utility.py [hunk 3]
```diff
@@ -13,3 +16,80 @@ def AddSystematicShift(
...
+            output.extend(CollectProducerOutput(prod))
+    return output
+
+
+class ProducerRule:
+    def __init__(self, producers, samples, scopes, invert=False, update_output=True):
+        self.producers = producers
+        self.samples = samples
+        self.scopes = scopes
+        self.invert = invert
```
Comment (harrypuuter) [inline config/utility.py:65]: What is this parameter needed for ?

File: config/utility.py [hunk 4]
```diff
@@ -13,3 +16,80 @@ def AddSystematicShift(
...
+        self.update_output = update_output
+
+    def applicable(self, sample):
+        applicable = sample in self.samples
+        if self.invert:
+            applicable = not applicable
+        return applicable
+
+    def operate(self, item, item_list):
```
Comment (harrypuuter) [inline config/utility.py:74]: so this does nothing yet ?

**Review by harrypuuter — CHANGES_REQUESTED** [review]
I have to say, i nave some trouble understanding, how this is supposed for work. Maybe the code just needs some more documentation ? The era and sample information will be added to the build_config function i guess ? But there is nothing there yet

File: config/utility.py (hunk 3 above)
Comment (swozniewski) [reply config/utility.py:65]: inversion of the sample selection

File: config/utility.py (hunk 2 above)
Comment (swozniewski) [reply config/utility.py:68]: checks whether modification should be applied based on samples and inversion flag

File: config/utility.py (hunk 4 above)
Comment (swozniewski) [reply config/utility.py:74]: only in the derived classes

Comment (swozniewski) [reply config/utility.py:74]: only for doc that this should be defined in derived classes

**Review by harrypuuter — APPROVED** [review]

**Pull request merged**